great_expectations/core/validation.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from great_expectations._docs_decorators import public_api
+from great_expectations.compatibility.pydantic import BaseModel
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch_config import BatchConfig
+    from great_expectations.core.expectation_suite import ExpectationSuite
+    from great_expectations.datasource.fluent.interfaces import DataAsset
+
+
+class Validation(BaseModel):
+    """
+    Responsible for running a suite against data and returning a validation result.
+
+    Args:
+        name: The name of the validation.
+        data: An asset or batch config to validate.
+        suite: A grouping of expectations to validate against the data.
+
+    """
+
+    name: str
+    data: DataAsset | BatchConfig
```

Comment:
I know we've talked about this some, but did we land on this being a union with DataAsset? I know it can be done with discriminated unions, but do we currently have a good story around serialization there? I'm a bit concerned about the ergonomics of accessing `validation.data` and having to do isinstance checks. Did we rule out a BatchConfig that represents the whole DataAsset?

Reply:
We could make this a BatchConfig and use pydantic coercion to allow a user to instantiate one with an asset.

Reply:
We've decided to define a protocol and allow this to be a union.